mitodl/open-discussions
# Added backend support for adding/deleting user websites

#### Pre-Flight checklist


#### What are the relevant tickets?
Provides the back end fro #1398 (a separate PR for the UI changes will be opened)

#### What's this PR do?
- Creates `UserWebsite` model
- Provides views for adding/deleting user websites

#### How should this be manually tested?
- Check that you can add and delete user websites via HTTP requests
- Check that you can't add multiple websites of the same 'type' for a user (e.g.: you can't submit 2 different sites with a 'facebook' domain, you can't submit 2 different sites with a domain that doesn't belong to one of our supported social sites since they will both be considered personal sites)
- Load your user's profile page and check the network tab - if you have created some user websites, they should be included in the `/api/v0/profiles/<your_username>/` response data

Code samples for the above using a Django shell:
```python
USER_EMAIL='<YOUR_USER_EMAIL_HERE>'
import json
from django.urls import reverse
from django.contrib.auth.models import User
from django.test.client import Client
user = User.objects.get(email=USER_EMAIL)
profile = user.profile
client = Client()
client.force_login(user)

# Create a new user website
websites_url = reverse('user_websites_api-list')
post_data = dict(
    profile=profile.id,
    url='http://www.facebook.com/someone'
)
resp = client.post(websites_url, data=json.dumps(post_data), content_type='application/json')

# Delete a user website
USER_WEBSITE_ID_TO_DELETE = 3
website_delete_url = reverse('user_websites_api-detail', args=[USER_WEBSITE_ID_TO_DELETE])
resp = client.delete(website_delete_url)
```
